core/src/main/scala/org/dbpedia/spotlight/db/memory/MemoryVectorStore.scala

```diff
@@ -0,0 +1,105 @@
+package org.dbpedia.spotlight.db.memory
+
+import breeze.linalg.{DenseVector, Transpose, DenseMatrix}
+import com.esotericsoftware.kryo.io.{Output, Input}
+import com.esotericsoftware.kryo.{Kryo, KryoSerializable}
+import org.dbpedia.spotlight.model.{TokenType, DBpediaResource}
+
+/**
+ * Created by dowling on 09/07/15.
+ */
+@SerialVersionUID(1008001)
+class MemoryVectorStore extends MemoryStore with KryoSerializable{
+
+  @transient
+  var vectors: DenseMatrix[Float] = null
+
+  @transient
+  var resourceIdToVectorIndex: Map[Int, Int] = null
+
+  @transient
+  var tokenTypeIdToVectorIndex: Map[Int, Int] = null
+
+  override def size: Int = vectors.rows
+
+  def _lookup(id: Int) = {
```

Comment:
these function definitions seem a bit pythonic in style to me, you should use the scala's camel case and probably change the iiner _lookup so some other name `lookupItem` or something